File: back/src/models/person.model.ts
// src/models/person.model.ts
import { Model, DataTypes } from 'sequelize';
import sequelize  from '../config/mysql';

class Person extends Model {
  public PersonID!: number;
  public FirstName!: string;
  public LastName!: string;
  public Address!: string;
  public Phone!: string;
  public Email!: string;
}

Person.init(
  {
    PersonID: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
    },
    FirstName: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    LastName: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    Address: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    Phone: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    Email: {
      type: DataTypes.STRING,
      allowNull: true,
    }
  },
  {
    sequelize,
    modelName: 'Person',
    tableName: 'Person',
    timestamps: false
  }
);

export default Person;

File: front/lib/model/person_model.dart
class Person {
  final int personID;
  final String firstName;
  final String lastName;
  final String address;
  final String phone;
  final String email;

  Person({
    required this.personID,
    required this.firstName,
    required this.lastName,
    required this.address,
    required this.phone,
    required this.email,
  });

  factory Person.fromJson(Map<String, dynamic> json) {
    return Person(
      personID: json['personID'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      address: json['address'],
      phone: json['phone'],
      email: json['email'],
    );
  }
}
